packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -225,40 +243,38 @@ export class WaferMap<
      * The updates snowball one after the other, this function only choses the 'altitude'.
      * The hover does not require an event update, but it's also the last update in the sequence.
      */
-    public update(): void {
+    public async update(): Promise<void> {
+        this.zoomHandler.connect();
```

Comment:
Why is calling `zoomHander.connect()` the right thing to do on _every_ update? Doesn't this add an event listener on every call? This seems like another case where a class should be leveraging notifiers in order for it to update as needed. For the `ZoomHandler`, it seems like it should be observing the `WaferMap`'s `canvasWidth`, `canvasHeight`, and `transform` properties, and when they change, calling `createZoomBehavior()`.

If this is deemed correct, but too large to handle for this PR (as it's tangential), I think we need another issue filed against this specifically.

Reply:
So we have 3 areas that the user might keep its mouse on.
![image](https://github.com/ni/nimble/assets/110180309/4ea3514a-13f2-4f02-9b16-76a04b6e37db)
Leaving `zoomHander.connect()` called on every update will cover all the areas. Trying to narrow the cases when zoom is connected while maintaining a similar behavior I arrived to the following:
Aria 1 - Can move `zoomHander.connect()` inside `hoverDieChanged()` observer, this leads to an unpleasant behavior as the user is able to zoom in for the first ~1 second while hovering after which if the user keep the mouse on the same die the observer will not be called again leading to not being able to zoom in.
Aria 2 - We don't have observers to track this area, though it doesn't contain any die I image a case when the user tries to zoom in near the corner and may have a hard time hovering a die considering we expect wafer up to 300K dies
Aria 3 - Don't think it's relevant to zoom

Calling `this.hoverHandler.disconnect()` at every `update()` makes the things complicated, though we still want to still want to delete the event listener.
Will leave the discussion open, @munteannatan any opinions?

Reply:
I dicussed with @munteannatan and moving both disconnect and connect inside the first if preserves the behavior 
![image](https://github.com/ni/nimble/assets/110180309/a49d455b-6314-4e04-ae26-f096b18670b0)

Reply:
I tried to implement your feedback about observables, now canvasWidthChanged() and canvasHeightChanged() handles zooming
![image](https://github.com/ni/nimble/assets/110180309/874406e2-fd4e-4400-b594-0be1c3adb7ba)

Reply:
So, we talked off-line about this and @Razvan1928 will try and implement the `Notifier` pattern in the `ZoomHandler` model so that the `WaferMap` doesn't have to call `connect` and `disconnect` directly in this way.

Reply:
Implemented!